### PR TITLE
misc(compare-runs): category scores, url filter, sort abs.val

### PR DESCRIPTION
**Pull request by connorjclark**
More stuff for compare runs script (related: #10519)

### CODE REVIEW

File: lighthouse-core/scripts/compare-runs.js [hunk 1]
```diff
@@ -205,14 +213,24 @@ function aggregateResults(name) {
   for (const lhrPath of glob.sync(`${outputDir}/**/lhr-*.json`)) {
     const lhrJson = fs.readFileSync(lhrPath, 'utf-8');
     /** @type {LH.Result} */
-    const lhr = JSON.parse(lhrJson);
+    const lhr = lhrJson && JSON.parse(lhrJson);
+    if (!lhr || !lhr.audits) {
+      console.warn(`could not parse ${lhrPath}`);
```
Comment (exterkamp) [inline lighthouse-core/scripts/compare-runs.js:218]: This isn't strictly true, if `!lhr` was false, but `!lhr.audits` was true, this just implies that there were no `audits` parsed. It still may have been parsed fine, might just not have audits. This error might not be specific enough to care the diff, but it is misleading.

File: lighthouse-core/scripts/compare-runs.js [hunk 2]
```diff
@@ -205,14 +213,24 @@ function aggregateResults(name) {
...
+      continue;
+    }
 
-    const metrics = lhr.audits.metrics ?
+    if (argv.urlFilter && !lhr.requestedUrl.includes(argv.urlFilter)) continue;
+
+    const metrics = lhr.audits.metrics && lhr.audits.metrics.details ?
```
Comment (exterkamp) [inline lighthouse-core/scripts/compare-runs.js:224]: This is starting to get out of hand. `&&` and a ternary in 1 assignment 😟 

Am I too go-thonic to say 
```js
const metrics = {}
if (lhr.audits.metrics && lhr.audits.metrics.details) {
    metrics = /** @type {!LH.Audit.Details.Table} */ (lhr.audits.metrics.details).items[0];
}
```

Honestly curious if it's more readable one way or the other. I think once a ternary becomes complex at all it's time to separate it out.

File: lighthouse-core/scripts/compare-runs.js [hunk 3]
```diff
@@ -205,14 +213,24 @@ function aggregateResults(name) {
...
     /** @type {!LH.Audit.Details.Table} */ (lhr.audits.metrics.details).items[0] :
       {};
     const allEntries = {
       metric: Object.entries(metrics).filter(([name]) => !name.endsWith('Ts')),
       timing: lhr.timing.entries.map(entry => ([entry.name, entry.duration])),
+      categories: Object.values(lhr.categories).reduce((acc, cur) => {
+        acc.push([cur.id, (cur.score || 0) * 100]);
+        return acc;
+      }, /** @type {Array<[string, number]>} */ ([])),
```
Comment (exterkamp) [inline lighthouse-core/scripts/compare-runs.js:233]: Seems weird to output this as an array of `strings` and `numbers`? And not as a map of `strings`:`numbers`? Any reason to output like this?

File: lighthouse-core/scripts/compare-runs.js (hunk 1 above)
Comment (connorjclark) [inline lighthouse-core/scripts/compare-runs.js:218]:
```suggestion
      console.warn(`lhr not found at ${lhrPath}`);
```

File: lighthouse-core/scripts/compare-runs.js (hunk 3 above)
Comment (connorjclark) [reply lighthouse-core/scripts/compare-runs.js:233]: The `allEntries` object expects this format. Did you see how it's used in the following loop, L241? 

The format comes naturally from `Object.entries(metrics) -> Array<[kind: string, value: number]>`

File: lighthouse-core/scripts/compare-runs.js (hunk 2 above)
Comment (connorjclark) [reply lighthouse-core/scripts/compare-runs.js:224]: if only `lhr.audits.metrics?.details` ...

> Am I too go-thonic to say

i think it's gooey :P I would prefer it if the ternary were more complex, but as it is it's just a nullability check.

**Pull request merged**